plugin/evm/validators_normalizer_test.go

```diff
@@ -0,0 +1,88 @@
+package evm
+
+import (
+	"math"
+	"math/rand"
+	"reflect"
+	"testing"
+
+	"github.com/flare-foundation/flare/ids"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestValidatorsNormalizer_ByEpoch(t *testing.T) {
+	var (
+		idsList []ids.ShortID
+		weights []uint64
+
+		normTestData     = testData{}
+		copyNormTestData = testData{}
+	)
+
+	numValidators := 6
+	var epochs = [2]uint64{0, 1}
+	for i := 0; i < numValidators; i++ {
+		idsList = append(idsList, randID())
+		weights = append(weights, rand.Uint64())
+
+		if i >= numValidators/len(epochs) {
+			normTestData[epochs[0]] = map[ids.ShortID]uint64{
+				idsList[i]: weights[i],
+			}
+			copyNormTestData[epochs[0]] = map[ids.ShortID]uint64{
+				idsList[i]: weights[i],
+			}
+			continue
+		}
+
+		normTestData[epochs[1]] = map[ids.ShortID]uint64{
+			idsList[i]: weights[i],
+		}
+		copyNormTestData[epochs[1]] = map[ids.ShortID]uint64{
+			idsList[i]: weights[i],
+		}
+	}
+
+	validatorsCache := NewValidatorsCache(normTestData, WithCacheSize(128))
+	for k, v := range normTestData {
+		validatorsCache.cache.Add(k, v)
+	}
+	normalizer := NewValidatorsNormalizer(validatorsCache)
+
+	t.Run("NonExistingKey", func(t *testing.T) {
+		res, err := normalizer.ByEpoch(uint64(7))
+		require.NoError(t, err)
+		assert.Len(t, res, 0)
+	})
+
+	t.Run("ExistingKey", func(t *testing.T) {
+		for _, epoch := range epochs {
+			result, err := normalizer.ByEpoch(epoch)
+			require.NoError(t, err)
+
+			var weightedNTData = map[ids.ShortID]uint64{}
+			for id, _ := range result {
+				weightedNTData[id] = copyNormTestData[epoch][id]
+			}
```

Comment:
I'm not fond of this solution tbh. If the point is to check that the weight ratio is the same, we can just compute it in advance and define it as a variable at the top of the test, and then use it only for the assertion. WDYT?

Reply:
We are not checking if the weight ratio is the same.
We are taking the old ratio to make parallel calculations that match the ones the `ByEpoch` method is doing, in order to
arrive at the same result of _some_ number of map values